cmd/crossplane/core/core.go

```diff
@@ -298,17 +301,46 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 		// fleshed out, implement a registration pattern similar to scheme
 		// registrations.
 		if err := xrd.SetupWebhookWithManager(mgr, o); err != nil {
-			return errors.Wrap(err, "cannot setup webhook for compositeresourcedefinitions")
+			return errors.Wrap(err, "Cannot setup webhook for compositeresourcedefinitions")
 		}
 		if err := composition.SetupWebhookWithManager(mgr, o); err != nil {
-			return errors.Wrap(err, "cannot setup webhook for compositions")
+			return errors.Wrap(err, "Cannot setup webhook for compositions")
 		}
 		if o.Features.Enabled(features.EnableAlphaUsages) {
 			if err := usage.SetupWebhookWithManager(mgr, o); err != nil {
-				return errors.Wrap(err, "cannot setup webhook for usages")
+				return errors.Wrap(err, "Cannot setup webhook for usages")
 			}
 		}
 	}
 
+	if err := c.SetupProbes(mgr); err != nil {
+		return errors.Wrap(err, "Cannot setup probes")
+	}
+
 	return errors.Wrap(mgr.Start(ctrl.SetupSignalHandler()), "Cannot start controller manager")
 }
+
+// SetupProbes sets up the health and ready probes.
+func (c *startCommand) SetupProbes(mgr ctrl.Manager) error {
+	// Add default readiness probe
+	if err := mgr.AddReadyzCheck("ping", healthz.Ping); err != nil {
+		return errors.Wrap(err, "cannot create ping ready check")
+	}
+
+	// Add default health probe
+	if err := mgr.AddHealthzCheck("ping", healthz.Ping); err != nil {
+		return errors.Wrap(err, "cannot create ping health check")
+	}
+
+	// Add probes waiting for the webhook server if webhooks are enabled
```

Comment:
Is there a way to add something like a started checker for the controllers? 
Could it be helpful to report as ready only after controllers are operational, e.g. after acquiring the lease successfully.

Reply:
We could also check/do that only after we're aware of a use case that it would help. 
Feel free to ignore for now.

Reply:
Yeah, we could definitely do something, but I couldn't find many examples of other projects doing much more than this. Acquiring the lease could not be the right condition though as we want to support multiple replicas serving webhooks, @sttts was talking about caches being populated, any additional hint/pointer to prior art?

Reply:
I don't think readiness is connected to leader election. It's a blackbox property whether the pod is ready to serve traffic (via the webhook). Controllers are always async. No need to wait for them. What does waiting for them even mean?

Reply:
Agree, @sttts. What did you have in mind by "waiting for informers" [here](https://crossplane.slack.com/archives/CEF5N8X08/p1696500988710769?thread_ts=1696496917.722149&channel=CEF5N8X08&message_ts=1696500988.710769)? I was referring to that when I asked if you had any prior reference implementation to share.

Reply:
If functionality of the process depends on informers to sync, then they should be part of readiness. E.g. admission in kube cannot work correctly without synced informers. Hence, the kube-apiserver might return inconsistent results prior to that.